packages/snap/src/index.ts

```diff
@@ -33,13 +19,28 @@ export type RequestParams = {
  * @throws If the request method is not valid for this snap, or if the request params are invalid. 
  */
 export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) => {
-  const params = request.params as unknown as RequestParams  // TODO improve type safety
+  const params = request.params as unknown as typeof RequestParams   
   const network = await getNetworkConfig()
   const provider = new CeloProvider(network.url)
   const bip44Node = await getBIP44Node()
   const privateKey = await getPrivateKey(bip44Node)
   const wallet = new CeloWallet(privateKey).connect(provider)
 
+  if (!RequestParams.is(params)) {
+    await snap.request({
+      method: 'snap_dialog',
+      params: {
+        type: 'alert',
+        content: panel([
+          text(`Invalid Request!`),
+          text(`${JSON.stringify(params.tx)}`)
+        ])
+      }
+    })
+
+    return;
+  }
+  
   switch (request.method) {
 
     case 'celo_sendTransaction':
```

Comment:
oh this is old but why are we using a special method here? 

alternatively we could be looking at the transaction type and if it matches `0x7c` then do the fee currency stuff.

Reply:
Not sure I'm following. What's 0x7c and what's the special method you're referring to?

Reply:
ok i understand my assumptions were incorrect